tools/distpartitioning/data_shuffle.py

```diff
@@ -155,38 +156,83 @@ def exchange_edge_data(rank, world_size, num_parts, edge_data):
 
     # Prepare data for each rank in the cluster. 
     start = timer()
-    for local_part_id in range(num_parts//world_size):
 
-        input_list = []
-        for idx in range(world_size):
-            send_idx = (edge_data[constants.OWNER_PROCESS] == (idx + local_part_id*world_size))
-            send_idx = send_idx.reshape(edge_data[constants.GLOBAL_SRC_ID].shape[0])
-            filt_data = np.column_stack((edge_data[constants.GLOBAL_SRC_ID][send_idx == 1], \
-                                    edge_data[constants.GLOBAL_DST_ID][send_idx == 1], \
-                                    edge_data[constants.GLOBAL_TYPE_EID][send_idx == 1], \
-                                    edge_data[constants.ETYPE_ID][send_idx == 1], \
-                                    edge_data[constants.GLOBAL_EID][send_idx == 1]))
-            if(filt_data.shape[0] <= 0):
-                input_list.append(torch.empty((0,5), dtype=torch.int64))
-            else:
-                input_list.append(torch.from_numpy(filt_data))
-
-        dist.barrier ()
-        output_list = alltoallv_cpu(rank, world_size, input_list, retain_nones=False)
-
-        #Replace the values of the edge_data, with the received data from all the other processes.
-        rcvd_edge_data = torch.cat(output_list).numpy()
-        edge_data[constants.GLOBAL_SRC_ID+"/"+str(local_part_id)] = rcvd_edge_data[:,0]
-        edge_data[constants.GLOBAL_DST_ID+"/"+str(local_part_id)] = rcvd_edge_data[:,1]
-        edge_data[constants.GLOBAL_TYPE_EID+"/"+str(local_part_id)] = rcvd_edge_data[:,2]
-        edge_data[constants.ETYPE_ID+"/"+str(local_part_id)] = rcvd_edge_data[:,3]
-        edge_data[constants.GLOBAL_EID+"/"+str(local_part_id)] = rcvd_edge_data[:,4]
+    CHUNK_SIZE = 100 * 1000 * 1000 # 100 * 8 * 5 = 1 * 4 = 8 GB/message/node 
+    num_edges = edge_data[constants.GLOBAL_SRC_ID].shape[0]
+    all_counts = allgather_sizes([num_edges], world_size, num_parts, return_sizes=True)
+    max_edges = np.amax(all_counts)
+    all_edges = np.sum(all_counts)
+    num_chunks = (max_edges // CHUNK_SIZE) + (0 if (max_edges % CHUNK_SIZE == 0) else 1)
```

Comment:
`math.ceil()` or `numpy.ceil()`?

Reply:
Agreed with this suggestion. 
Will make this change in the next iteration of this PR, In which I intend to perform numpy optimizations for the unique function. This PR has been sitting in the queue for various reasons for the past 3 weeks (either for lintrunner reasons or for some other reasons). I will commit this PR for now.